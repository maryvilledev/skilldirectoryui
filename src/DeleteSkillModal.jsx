import React from 'react'
import Button from 'react-bootstrap/lib/Button'

class DeleteSkillModal extends React.Component {
  constructor(){
    super();
    this.confirm = this.confirm.bind(this);
  }
  render() {
    return (
      <div>
        <h1>Are you sure?</h1>
        <Button onClick={this.confirm} name="Yes">Yes</Button>
        <Button onClick={this.props.closeModal} name="No">No</Button>
      </div>
    );
  }
  confirm() {
    this.props.doDelete();
    this.props.closeModal();
  }
}

export default DeleteSkillModal
