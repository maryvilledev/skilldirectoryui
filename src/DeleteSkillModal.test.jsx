import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import DeleteSkillModal from './DeleteSkillModal';

describe('<DeleteSkillModal />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<DeleteSkillModal />, div);
  });

  it('matches the stored snapshot', () => {
    const wrapper = shallow(<DeleteSkillModal />);
    expect(wrapper.get(0)).toMatchSnapshot();
  });

  it('has a yes button that triggers delete callback and closes modal', () => {
    let didDelete = false;
    let didClose = false;
    const mockDelete = function() {didDelete = true;};
    const mockClose = function() {didClose = true;};
    const wrapper = mount(<DeleteSkillModal
      doDelete={mockDelete}
      closeModal={mockClose}
    />);
    const button = wrapper.find("button")
      .findWhere(n => n.debug().includes('name="Yes"'));
    button.simulate('click');
    expect(didDelete && didClose).toBe(true);

  });

  it('has a no button that doesn\'t trigger delete callback but closes modal',
   () => {
    let didDelete = false;
    let didClose = false;
    const mockDelete = function() {didDelete = true;};
    const mockClose = function() {didClose = true;};
    const wrapper = mount(<DeleteSkillModal
      doDelete={mockDelete}
      closeModal={mockClose}
    />);
    const button = wrapper.find("button")
      .findWhere(n => n.debug().includes('name="No"'));
    button.simulate('click');
    expect(didClose).toBe(true);
    expect(didDelete).toBe(false);
  });
});
