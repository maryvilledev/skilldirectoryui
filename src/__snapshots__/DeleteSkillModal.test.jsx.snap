exports[`<DeleteSkillModal /> matches the stored snapshot 1`] = `
<div>
  <h1>
    Are you sure?
  </h1>
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="default"
    disabled={false}
    name="Yes"
    onClick={[Function]}>
    Yes
  </Button>
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="default"
    disabled={false}
    name="No"
    onClick={undefined}>
    No
  </Button>
</div>
`;
